docs/basic-features/data-fetching/getStaticProps.md

```diff
@@ -0,0 +1,151 @@
+---
+description: Fetch data at build time with `getStaticProps` (Static Generation) API reference.
+---
+
+# `getStaticProps`
+
+If you export an `async` function called `getStaticProps` (static generation) from a page, Next.js will pre-render this page at build time using the props returned by `getStaticProps`.
+
+```jsx
+export async function getStaticProps(context) {
+  return {
+    props: {}, // will be passed to the page component as props
+  }
+}
+```
+
+## When should I use `getStaticProps`?
+
+You should use `getStaticProps` if:
+
+- The data required to render the page is available at build time ahead of a user’s request.
+- The data comes from a headless CMS.
+- The data can be publicly cached (not user-specific).
+- The page must be pre-rendered (for SEO) and be very fast — `getStaticProps` generates `HTML` and `JSON` files, both of which can be cached by a CDN for performance.
```

Comment:
Hmmm actually I like this - let's put it in getStaticPaths too?

Reply:
Maybe also move this to intro of getStaticProps instead of the end?

Reply:
@Nutlope this is right near the top of the page. 😊

Reply:
Which part @leerob? All the bullets?

Reply:
Yeah they're all good. Only needs a slight tweak on "build time" as mentioned in other comments. We could also say headless CMS or database to be more wide 😄